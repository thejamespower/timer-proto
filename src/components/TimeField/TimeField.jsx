import React from 'react'
import TimeField from 'react-simple-timefield'
import PropTypes from 'prop-types'

import { yellow800 } from 'material-ui/styles/colors'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import IconClock from 'material-ui/svg-icons/device/access-time'
import TextField from 'material-ui/TextField'

export default class CustomTimeField extends React.Component {
  static propTypes = {
    onChange: PropTypes.func,
    value: PropTypes.string,
  }

  static defaultProps = {
    onChange: null,
    value: null,
  }

  constructor(...args) {
    super(...args)

    this.state = {
      value: args.value || '00:00:00',
    }

    this.onChange = this.onChange.bind(this)
  }

  componentWillReceiveProps(nextProps) {
    const { value: nextValue } = nextProps
    const { value: stateValue } = this.state
    if (nextValue !== undefined && nextValue !== stateValue) {
      this.setState({ value: nextValue })
    }
  }

  onChange(time) {
    const { value } = this.state
    const newTime = time.replace(/-/g, ':')
    const newTimeSeconds = newTime.padEnd(8, value.substr(5, 3))

    this.setState({ value: newTimeSeconds })
  }

  render() {
    const { value } = this.state

    const muiTheme = getMuiTheme({
      fontFamily: 'Arial',
      palette: {
        primary1Color: yellow800,
      },
      textField: {
        floatingLabelColor: '#666',
      },
    })

    const { onChange = this.onChange } = this.props

    return (
      <section className="container">
        <MuiThemeProvider muiTheme={muiTheme}>
          <div style={{ marginRight: 20 }}>
            <IconClock style={{ width: 25, marginRight: 6, marginBottom: -6 }} color="#bbb" />
            <TimeField
              showSeconds
              value={value}
              onChange={onChange}
              style={{ width: 82, fontSize: 20 }}
              input={<TextField floatingLabelFixed floatingLabelText="Time" />}
            />
          </div>
        </MuiThemeProvider>
      </section>
    )
  }
}
