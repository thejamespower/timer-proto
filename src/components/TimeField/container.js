import { connect } from 'react-redux'

import TimeField from './TimeField'

const mapStateToProps = null

const mapDispatchToProps = null

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TimeField)
