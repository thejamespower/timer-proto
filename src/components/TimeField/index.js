import TimeField from './container'

export default TimeField
