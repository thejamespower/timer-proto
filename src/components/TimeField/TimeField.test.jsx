import React from 'react'
import renderer from 'react-test-renderer'
import TimeField from './TimeField'

describe('TimeField', () => {
  const props = {
    onChange: () => null,
    value: '',
  }

  it('renders', () => {
    const component = renderer.create(
      <TimeField {...props} />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
