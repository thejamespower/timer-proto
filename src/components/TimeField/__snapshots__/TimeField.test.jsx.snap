// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeField renders 1`] = `
<section
  className="container"
>
  <div
    style={
      Object {
        "marginRight": 20,
      }
    }
  >
    <svg
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fill": "#bbb",
          "height": 24,
          "marginBottom": -6,
          "marginRight": 6,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "width": 25,
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
      />
    </svg>
    <div
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Arial",
          "fontSize": 20,
          "height": 72,
          "lineHeight": "24px",
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 82,
        }
      }
    >
      <label
        htmlFor="undefined-undefined-Time-2642"
        style={
          Object {
            "color": "#666",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(0.75) translate(0, -28px)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }
      >
        Time
      </label>
      <input
        disabled={false}
        id="undefined-undefined-Time-2642"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitOpacity": 1,
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "appearance": "textfield",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "font": "inherit",
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        type="text"
        value="00:00:00"
      />
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 2,
              "borderColor": "#f9a825",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</section>
`;
